drinfomon.lic

```diff
@@ -340,6 +335,8 @@ class DRSkill
       skill.exp = exp.to_i
       skill.percent = percent.to_i
       skill.current = rank.to_i + (percent.to_i / 100.0)
+    else
+      DRSkill.new(name, rank, exp, percent)
```

Comment:
Yeah, it's a strange paradigm for sure. The constructor creates a new instance **and then** adds it to a `@@list` class variable. So at this point, we don't care about the return value because the constructor is already adding the reference to a list that's available class-wide.

```ruby
  def self.update(name, rank, exp, percent)
    skill = self.find_skill(name) # this returns the record from `@@list`
    if skill
      # updates the variable properties
    else
      DRSkill.new(name, rank, exp, percent) # adds a new instance to `@@list`
    end
end
```